adapters/handlers/grpc/v1/mapping.go

```diff
@@ -192,23 +182,222 @@ func NewObject[P schema.PropertyInterface](v map[string]interface{}, parent P, s
+		}
+	}
+	return &pb.Value{Kind: &pb.Value_ListValue{ListValue: listValue}}, nil
+}
+
+func (m *Mapper) newListValueInt(v interface{}) (*pb.Value, error) {
+	var listValue *pb.ListValue
+	if _, ok := v.([]interface{}); ok {
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_IntValues{IntValues: &pb.IntValues{Values: []int64{}}}}
+		} else {
+			listValue = &pb.ListValue{Values: []*pb.Value{}}
+		}
+	} else {
+		values, err := parseArray[float64](v, schema.DataTypeIntArray)
+		if err != nil {
+			return nil, err
+		}
+		if m.uses125 {
+			parsedValues := make([]int64, len(values))
+			for i, v := range values {
+				parsedValues[i] = int64(v)
+			}
+			listValue = &pb.ListValue{Kind: &pb.ListValue_IntValues{IntValues: &pb.IntValues{Values: parsedValues}}}
+		} else {
+			x := make([]*pb.Value, len(values))
+			for i, v := range values {
+				x[i] = NewIntValue(int64(v))
+			}
+			listValue = &pb.ListValue{Values: x}
+		}
+	}
+	return &pb.Value{Kind: &pb.Value_ListValue{ListValue: listValue}}, nil
+}
+
+func (m *Mapper) newListValueString(v interface{}) (*pb.Value, error) {
+	var listValue *pb.ListValue
+	if _, ok := v.([]interface{}); ok {
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_TextValues{TextValues: &pb.TextValues{Values: []string{}}}}
+		} else {
+			listValue = &pb.ListValue{Values: []*pb.Value{}}
+		}
+	} else {
+		values, err := parseArray[string](v, schema.DataTypeStringArray)
+		if err != nil {
+			return nil, err
+		}
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_TextValues{TextValues: &pb.TextValues{Values: values}}}
+		} else {
+			x := make([]*pb.Value, len(values))
+			for i, v := range values {
+				x[i] = NewStringValue(v)
+			}
+			listValue = &pb.ListValue{Values: x}
+		}
+	}
+	return &pb.Value{Kind: &pb.Value_ListValue{ListValue: listValue}}, nil
+}
+
+func (m *Mapper) newListValueText(v interface{}) (*pb.Value, error) {
+	var listValue *pb.ListValue
+	if _, ok := v.([]interface{}); ok {
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_TextValues{TextValues: &pb.TextValues{Values: []string{}}}}
+		} else {
+			listValue = &pb.ListValue{Values: []*pb.Value{}}
+		}
+	} else {
+		values, err := parseArray[string](v, schema.DataTypeTextArray)
+		if err != nil {
+			return nil, err
+		}
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_TextValues{TextValues: &pb.TextValues{Values: values}}}
+		} else {
+			x := make([]*pb.Value, len(values))
+			for i, v := range values {
+				x[i] = NewStringValue(v)
+			}
+			listValue = &pb.ListValue{Values: x}
+		}
+	}
+	return &pb.Value{Kind: &pb.Value_ListValue{ListValue: listValue}}, nil
+}
+
+func (m *Mapper) newListValueUuid(v interface{}) (*pb.Value, error) {
+	var listValue *pb.ListValue
+	if _, ok := v.([]interface{}); ok {
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_UuidValues{UuidValues: &pb.UuidValues{Values: []string{}}}}
```

Comment:
should we send them as bytes to avoid the costly parsing on the python side?

Reply:
I contemplated this. If we did it for one, should we do it for all? I'd be happy for this so that all `list_value` fields just send bytes over but with a different key for each type so that the clients know how to parse them appropriately

Reply:
I think we should test if the other arrays are also really slow in the GRPC code or if that was only the case for floats - I can't remember if we only noticed because of the float-vectors

Reply:
vectors were always pure float lists:
```
  repeated float vector = 2 [deprecated = true]; // deprecated, will be removed
```

To make it a bit faster: Just add a property with int/string etc lists and see if it shows up in profiling